### ISSUE TITLE
loss.backward ：gradient cannot be calculated

### ISSUE THREAD

**Issue by DD22333** [vivian1993/NCRL#8]
Hello,author! When I re-run the code, when the rule length is 3, an error occurs in loss.backward. A variable is modified in place and the gradient cannot be calculated. Is it possible that it is due to the version? (My environment is pytorch1.7+GPU). Therefore, I modified the situation where sseq_len is greater than 2 in the reduce_rel_pair function, especially the output operation, to avoid in-place operations, but it did not solve the problem. It still exists. Can you give me some helpful suggestions?Thanks. 

Specific error message:“ one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [50, 1, 25]], which is output 0 of DivBackward0,is at version 1; expected version 0 instead. Hint: the backtrace further above shows the operation that failed to compute its gradient. The variable in question was changed in thereor anywhere later. ”
